File: app/src/main/java/me/rohank05/beat/lavaplayer/GuildMusicManager.java
package me.rohank05.beat.lavaplayer;

import com.sedmelluq.discord.lavaplayer.player.AudioPlayer;
import com.sedmelluq.discord.lavaplayer.player.AudioPlayerManager;

import net.dv8tion.jda.api.entities.TextChannel;

public class GuildMusicManager {
    public final AudioPlayer audioPlayer;
    public final TrackScheduler scheduler;
    private final AudioPlayerSendHandler sendHandler;

    public GuildMusicManager(AudioPlayerManager manager, TextChannel channel) {
        this.audioPlayer = manager.createPlayer();
        this.scheduler = new TrackScheduler(this.audioPlayer, channel);
        this.audioPlayer.addListener(this.scheduler);
        this.sendHandler = new AudioPlayerSendHandler(this.audioPlayer);

    }

    public AudioPlayerSendHandler getSendHandler() {
        return this.sendHandler;
    }
}

File: src/main/kotlin/com/therohankumar/modules/GuildMusicManager.kt
package com.therohankumar.modules

import com.sedmelluq.discord.lavaplayer.player.AudioPlayer

class GuildMusicManager(audioPlayer: AudioPlayer){
    val sendHandler: AudioPlayerSendHandler = AudioPlayerSendHandler(audioPlayer)
    val trackScheduler = TrackScheduler(audioPlayer)
    val audioFilter = Filters(audioPlayer)
    val player = audioPlayer
    init {
        audioPlayer.addListener(trackScheduler)
    }
}